appsDiscovery/src/main/java/org/kinecosystem/appsdiscovery/sender/service/SendKinServiceBase.java

```diff
@@ -0,0 +1,69 @@
+package org.kinecosystem.appsdiscovery.sender.service;
+
+import android.app.Service;
+import android.content.Intent;
+import android.os.Binder;
+import android.os.IBinder;
+import android.support.annotation.NonNull;
+
+public abstract class SendKinServiceBase extends Service {
+
+    //Developer needs to implements
+    //these is called from background thread so can be called async
+    public abstract @NonNull
+    KinTransferComplete transferKin(@NonNull final String toAddress, final int amount, @NonNull final String memo) throws KinTransferException;
+
+    public abstract int getCurrentBalance() throws BalanceException;
```

Comment:
should return `BigDecimal` like in `kin-sdk`

Reply:
cool